tests/test_packages/test_contracts/test_erc1155/test_contract.py

```diff
@@ -232,3 +238,171 @@ def test_get_batch_atomic_swap(ledger_api, crypto_api, erc1155_contract):
             for key in ["value", "chainId", "gas", "gasPrice", "nonce", "to", "from"]
         ]
     ), "Error, found: {}".format(tx)
+
+
+class TestCosmWasmContract:
+    def setup(self):
+        self.ledger_api = ledger_apis_registry.make(COSMOS, **COSMOS_TESTNET_CONFIG)
+        self.faucet_api = faucet_apis_registry.make(COSMOS)
+        self.deployer_crypto = crypto_registry.make(COSMOS)
+        self.item_owner_crypto = crypto_registry.make(COSMOS)
+
+        self.token_ids_a = [
+            340282366920938463463374607431768211456,
+            340282366920938463463374607431768211457,
+            340282366920938463463374607431768211458,
+            340282366920938463463374607431768211459,
+            340282366920938463463374607431768211460,
+            340282366920938463463374607431768211461,
+            340282366920938463463374607431768211462,
+            340282366920938463463374607431768211463,
+            340282366920938463463374607431768211464,
+            340282366920938463463374607431768211465,
+        ]
+
+        self.token_id_b = 680564733841876926926749214863536422912
+
+        # Refill deployer account from faucet
+        self.refill_from_faucet(self.ledger_api, self.faucet_api, self.deployer_crypto.address)
+
+        # Refill item owner account from faucet
+        self.refill_from_faucet(self.ledger_api, self.faucet_api, self.item_owner_crypto.address)
+
+    def refill_from_faucet(self, ledger_api, faucet_api, address):
+        start_balance = ledger_api.get_balance(address)
+
+        faucet_api.get_wealth(address)
+
+        tries = 10
+        while tries > 0:
+            tries -= 1
+            time.sleep(1)
+
+            balance = ledger_api.get_balance(address)
+            if balance != start_balance:
+                break
+
+    @pytest.mark.integration
+    @pytest.mark.ledger
+    def test_cosmwasm_contract_deploy_and_interact(self, erc1155_contract):
+
+        # Deploy contract
+        tx = erc1155_contract.get_deploy_transaction(
+            ledger_api=self.ledger_api, deployer_address=self.deployer_crypto.address, gas=900000
+        )
+        assert len(tx) == 6
+        signed_tx = self.deployer_crypto.sign_transaction(tx)
+        receipt = json.loads(self.ledger_api.send_signed_transaction(signed_tx))
+        assert len(receipt) == 7
+        assert all(
+            [
+                key in receipt
+                for key in ["height", "txhash", "data", "raw_log", "logs", "gas_wanted", "gas_used"]
+            ])
+
+        code_id = self.ledger_api.get_last_code_id()
+
+        # Init contract
+        tx = self.ledger_api.get_init_transaction(self.deployer_crypto.address, code_id, {}, 0, 0, label="ERC1155")
```

Comment:
this needs to be requested from contract (which will then request it from ledger api); so call contract here

Reply:
(e.g. pattern should always be a) talk to contract; b) talk to crypto; c) talk to ledger api